store/mongo/datastore_mongo.go

```diff
@@ -567,11 +591,20 @@ func (db *DataStoreMongo) UpsertRemoveDeviceAttributes(
 		return nil, err
 	}
 
+	if scope == model.AttrScopeTags {
+		update[etagField] = uuid.New().String()
+	}
+
 	now := time.Now()
-	update[updatedField] = model.DeviceAttribute{
-		Scope: model.AttrScopeSystem,
-		Name:  model.AttrNameUpdated,
-		Value: now,
+	filter := bson.M{"_id": id}
+	if etag != "" {
+		filter[etagField] = bson.M{"$eq": etag}
+	} else {
+		update[updatedField] = model.DeviceAttribute{
+			Scope: model.AttrScopeSystem,
+			Name:  model.AttrNameUpdated,
+			Value: now,
+		}
```

Comment:
This is wrong, I believe: the ETag is not mandatory, you cannot rely on it to ste the "updated at" field.
If I set tags but I do not provide the If-Match, you are still updating the timestamp.

Reply:
You are right, counting on etag here is wrong, scope sounds the right criteria.